idunn/api/instant_answer.py

```diff
@@ -267,53 +274,79 @@ async def fetch_pj_response():
         )
 
     # Query PJ API and Bragi osm asynchronously as a task which may be cancelled
-    # if bragi tripadvisor finds results.
     fetch_pj = asyncio.create_task(fetch_pj_response())
     fetch_bragi_osm = asyncio.create_task(bragi_client.autocomplete(query))
     bragi_tripadvisor_response = await bragi_client.autocomplete(query_tripadvisor)
     bragi_tripadvisor_features = result_filter.filter_bragi_features(
         normalized_query, bragi_tripadvisor_response["features"]
     )
 
-    for bragi_tripadvisor_feature in bragi_tripadvisor_features:
-        feature_properties = bragi_tripadvisor_feature["properties"]["geocoding"]
-        if (
-            "poi_types" in feature_properties
-            and feature_properties["poi_types"][0]["id"].split(":")[1] == "subclass_hotel"
-        ):
-            fetch_pj.cancel()
-            fetch_bragi_osm.cancel()
-            place_id = feature_properties["id"]
-            return await run_in_threadpool(
-                get_instant_answer_single_place,
-                query=q,
-                place_id=place_id,
-                lang=lang,
-                type="poi_tripadvisor",
-            )
-
+    # Select datasource instant answer in France
+    if len(intentions) > 0 and pj_source.bbox_is_covered(intentions[0].filter.bbox):
```

Comment:
I'm not sure I understand the reason for `len(intentions) > 0`. Maybe it is put here in order to check if the user is actually looking for an hotel?

Reply:
I check before accessing `intentions[0]` to prevent 'NoneType' AttributeError